datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -276,15 +278,74 @@ impl<'a> PruningStatistics for RowGroupPruningStatistics<'a> {
         scalar.to_array().ok()
     }
 
+    /// The basic idea is to check whether all of the `values` are not within the min-max boundary.
+    /// If any one value is within the min-max boundary, then this row group will not be skipped.
+    /// Otherwise, this row group will be able to be skipped.
     fn contained(
         &self,
-        _column: &Column,
-        _values: &HashSet<ScalarValue>,
+        column: &Column,
+        values: &HashSet<ScalarValue>,
     ) -> Option<BooleanArray> {
-        None
+        let min_values = self.min_values(column)?;
+        let max_values = self.max_values(column)?;
+        // The boundary should be with length of 1
+        if min_values.len() != max_values.len() || min_values.len() != 1 {
+            return None;
+        }
+        let min_value = ScalarValue::try_from_array(min_values.as_ref(), 0).ok()?;
+        let max_value = ScalarValue::try_from_array(max_values.as_ref(), 0).ok()?;
+
+        // The boundary should be with the same data type
+        if min_value.data_type() != max_value.data_type() {
+            return None;
+        }
+        let target_data_type = min_value.data_type();
+
+        let (c, _) = self.column(&column.name)?;
+        let has_null = c.statistics()?.null_count() > 0;
+        let mut known_not_present = true;
+        for value in values {
+            // If it's null, check whether the null exists from the statistics
```

Comment:
If `value` is null, I think it means that the statistics value is not known. To the best of my knowledge, `NULL` values on the column are never encoded in parquet statistics .

Thus I think this check needs to be something like 
```rust
if value.is_null() { 
  known_not_present = false;
  break;
```

Reply:
Hi @alamb, this case is for filters like `col is null`. It's not related to the statistics. The values are from the filter literals.

Reply:
I think `col in (NULL)` will not match any thing,  same as `col = null`, which means `col in (a,b,c)` same as `col in (a,b,c, null)`. is there any rule to remove the null out of in list 🤔 @alamb

Reply:
> I think `col in (NULL)` will not match any thing, same as `col = null`, which means `col in (a,b,c)` same as `col in (a,b,c, null)`. is there any rule to remove the `null` out of in list 🤔 @alamb

@Ted-Jiang  -- I think we discussed this in https://github.com/apache/arrow-datafusion/issues/8688